posthog/hogql_queries/hogql_cohort_query.py

```diff
@@ -438,6 +442,56 @@ def _get_condition_for_property(self, prop: Property) -> ast.SelectQuery | ast.S
     def _get_conditions(self) -> ast.SelectQuery | ast.SelectSetQuery:
         Condition = namedtuple("Condition", ["query", "negation"])
 
+        def unwrap_property(prop: Union[PropertyGroup, Property]) -> Optional[Property]:
+            """Unwrap a PropertyGroup to get the underlying Property if it contains exactly one."""
+            if isinstance(prop, Property):
+                return prop
+            if isinstance(prop, PropertyGroup) and len(prop.values) == 1:
+                return unwrap_property(prop.values[0])
+            return None
+
+        def can_combine_person_properties(properties: Union[list[PropertyGroup], list[Property]]) -> bool:
+            """Check if all properties are person properties that can be combined into a single query."""
```

Comment:
Handle empty properties edge case. If `properties` is empty, `all` would return `true` here.

```suggestion
            """Check if all properties are person properties that can be combined into a single query.""" 
            if not properties:  # Guard against empty lists
                return False
```